File: src/extension/storage.cpp
#include "storage.h"
#include <stdlib.h>
#include <string.h>

Storage::Storage() {

    this->pattern_len = 0;
    this->pattern = NULL;

    this->cost_ins = 1;
    this->cost_del = 1;
    this->cost_rep = 1;

    this->distance = 0;
    this->path = NULL;
    this->blocks = NULL;
    this->searches = NULL;

    this->lv = NULL;
    this->match_map = NULL;

}

File: src/Nfilin/Libs/Storage.php
<?php
namespace Nfilin\Libs;


/**
 * Class Storage
 * @package Nfilin\Libs
 */
class Storage
{
    /**
     * @var double
     */
    public $cost_ins = 1;
    /**
     * @var double
     */
    public $cost_rep = 1;
    /**
     * @var double
     */
    public $cost_del = 1;
    /**
     * @var string[]|null
     */
    public $pattern;
    /**
     * @var string[]|null
     */
    public $path;
    /**
     * @var string[]|null
     */
    public $blocks;
    /**
     * @var string[]|null
     */
    public $searches;
    /**
     * @var double[][]|null
     */
    public $lv;
    /**
     * @var int[][]|null
     */
    public $match_map;
    /**
     * @var double
     */
    public $distance = 0;
}